test/iron-dropdown.html

```diff
@@ -512,6 +526,50 @@
           });
         });
       });
+
+      suite('auto-cancel on blur', function() {
+
+        test('auto-cancel when focusable dropdown blurs', function(done) {
+          var elements = fixture('FocusableDropdownWithOtherFocusableElements');
+          var dropdown = Polymer.dom(elements).querySelector('iron-dropdown');
+          var content = Polymer.dom(dropdown).querySelector('.dropdown-content');
+          var button1 = Polymer.dom(elements).querySelector('#focusableButton1');
+          var button2 = Polymer.dom(elements).querySelector('#focusableButton2');
+
+          // Step 1: First button gets focus.
+          button1.addEventListener('focus', function() {
+            dropdown.open();
+            runAfterOpen(dropdown, function() {
+              // Opening the dropdown moves focus to the dropdown content, but
+              // not if the window doesn't have focus, so to cover that case we
+              // have to simulate a focus event.
+              MockInteractions.focus(content);
+            });
+          });
+
+          // Step 2: Dropdown content gets focus after opening.
+          content.addEventListener('focus', function() {
+            // Move focus outside the dropdown to trigger auto-cancel behavior.
+            // To cover both interactive and automated testing, we need to do
+            // both a real focus and simulated focus change.
+            button2.focus();
+            MockInteractions.focus(button2);
```

Comment:
For the test to work, the `activeElement` needs to change, so we need to really invoke `.focus()` on the button, not just dispatch a `focus` event on it. As written, `MockInteractions.focus()` just raises the event.

It's also not sufficient to just invoke `.focus()`, because that won't raise the `focus` event if the window is inactive.

Because this is a general problem with unit tests that deal with focus, I think it's worth generalizing the solution. I've replaced the hack of calling both real `focus` and mock `focus` with a helper `mockFocusAndEvent`. This invokes `focus()` and then, if that didn't automatically raise the `focus` event, simulates the event.

If that approach seems correct, I'd recommend folding this helper into `MockInteractions.focus`. That can be done separately in `iron-test-helpers`, after which this helper could be removed from the `iron-dropdown` tests. For now, I didn't want to hold up this already-complex PR with a PR on yet another repo.